module/zfs/zap_micro.c

```diff
@@ -54,14 +54,25 @@
  * machinery to understand not to try to split a microzap block).
  *
  * If large_microzap is enabled, this value will be clamped to
- * spa_maxblocksize(). If not, it will be clamped to SPA_OLD_MAXBLOCKSIZE.
+ * spa_maxblocksize(), up to 1M. If not, it will be clamped to
+ * SPA_OLD_MAXBLOCKSIZE.
  */
 static int zap_micro_max_size = SPA_OLD_MAXBLOCKSIZE;
 
+/*
+ * The 1M upper limit is necessary because the count of chunks in a microzap
+ * block is stored as an int64_t (zap_num_chunks). Each chunk is 64 bytes, and
```

Comment:
```suggestion
 * block is stored as an int16_t (zap_num_chunks). Each chunk is 64 bytes, and
```

Reply:
Heh thanks. My brain autocomplete gets "int six" and offers "four".